hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReliableTopicProxy.java

```diff
@@ -197,6 +213,125 @@ public LocalTopicStats getLocalTopicStats() {
         throw new UnsupportedOperationException("Locality is ambiguous for client!");
     }
 
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> payload) {
+        checkNotNull(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(payload, NULL_MESSAGE_IS_NOT_ALLOWED);
+
+        try {
+            List<ReliableTopicMessage> messages = payload.stream()
+                    .map(m -> new ReliableTopicMessage(toData(m), null))
+                    .collect(Collectors.toList());
+            switch (overloadPolicy) {
+                case ERROR:
+                    long sequenceId = ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).toCompletableFuture().get();
+                    if (sequenceId == -1) {
+                        throw new TopicOverloadException("Failed to publish messages: " + payload + " on topic:" + getName());
+                    }
+                    break;
+                case DISCARD_OLDEST:
+                    ringbuffer.addAllAsync(messages, OverflowPolicy.OVERWRITE).toCompletableFuture().get();
+                    break;
+                case DISCARD_NEWEST:
+                    ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).toCompletableFuture().get();
+                    break;
+                case BLOCK:
+                    long timeoutMs = INITIAL_BACKOFF_MS;
+                    for (; ; ) {
+                        long result = ringbuffer.addAllAsync(messages, OverflowPolicy.FAIL).toCompletableFuture().get();
```

Comment:
Just occurred to me - can you check if this addition is atomic - either all are added or none? Asking because then it's safe to retry but if we added some and not others, we might end up adding some items multiple times.

Reply:
Ok, I investigated this. The `com.hazelcast.ringbuffer.impl.RingbufferProxy#addAllAsync` method uses the `com.hazelcast.ringbuffer.impl.operations.AddAllOperation` class for adding messages. In this operation (please see the run method), a check about the remaining capacity is accomplished. So either every item is added to the ringbuffer or none.

Reply:
Great, thanks!